### PR TITLE
[kube-prometheus-stack] Minify Grafana Dashboards and Update CRDs to 0.65.1

### LINKED ISSUE #3214: [kube-prometheus-stack] update to 0.64
### Is your feature request related to a problem ?

The prometheus operator is now on version 0.64, it would be nice if this chart followed.

### Describe the solution you'd like.

update CRDs to 0.64

### Describe alternatives you've considered.

Using another chart

### Additional context.

https://github.com/prometheus-operator/prometheus-operator/releases/tag/v0.64.0

### PR DESCRIPTION
**Pull request by jkroepke**
#### What this PR does / why we need it

Reduce the helm chart size (see https://github.com/prometheus-community/helm-charts/pull/3267#issuecomment-1528349719)

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #3214
- replaces #3267

#### Special notes for your reviewer

Size of the helm release on branch main
```
% kubectl get secrets sh.helm.release.v1.kube-prometheus-stack-f1dkdbv43c.v1 -o jsonpath='{.data.release}' | base64 -d > release
% ls -lah release 
-rw-r--r--  1 jkr  staff   971K May 23 08:05 release
```

Size of the helm release on branch minify-json-dashboards
```
% kubectl get secrets sh.helm.release.v1.kube-prometheus-stack-f1dkdbv43c.v1 -o jsonpath='{.data.release}' | base64 -d > release      
% ls -lah release                                                                                                                     
-rw-r--r--  1 jkr  staff   979K May 23 08:07 release

```

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py [hunk 1]
```diff
@@ -120,49 +111,17 @@ def patch_dashboards_json(content, multicluster_key):
...
             overwrite_list.append(variable)
         content_struct['templating']['list'] = overwrite_list
-
-        # fix drilldown links. See https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/659
-        for row in content_struct['rows']:
-            for panel in row['panels']:
-                for style in panel.get('styles', []):
-                    if 'linkUrl' in style and style['linkUrl'].startswith('./d'):
-                        style['linkUrl'] = style['linkUrl'].replace('./d', '/d')
-
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:130]: this is fixed upstream

File: charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py [hunk 2]
```diff
@@ -120,49 +111,17 @@ def patch_dashboards_json(content, multicluster_key):
                 variable['hide'] = ':multicluster:'
             overwrite_list.append(variable)
         content_struct['templating']['list'] = overwrite_list
-
-        # fix drilldown links. See https://github.com/kubernetes-monitoring/kubernetes-mixin/issues/659
-        for row in content_struct['rows']:
-            for panel in row['panels']:
-                for style in panel.get('styles', []):
-                    if 'linkUrl' in style and style['linkUrl'].startswith('./d'):
-                        style['linkUrl'] = style['linkUrl'].replace('./d', '/d')
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:129]: This is fixed upstream

**Comment by QuentinBisson** [comment]
@jkroepke this is an awesome find for now :) I think we are missing the RBAC to access those CRDs but I will take a better look today

**Comment by jkroepke** [comment]
@QuentinBisson Maintainers are allowed to edit this pull request is enabled, so feel free to commit into this.

File: charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml [hunk 3]
```diff
@@ -11,14 +11,19 @@ rules:
   - monitoring.coreos.com
   resources:
```
Comment (jkroepke) [inline charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:12]: @QuentinBisson is it possible to generate this list? Or its an manual doing?

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus-operator/clusterrole.yaml:12]: It is done manually at the moment but we can probably have it done automatically as it's coming from https://github.com/prometheus-operator/kube-prometheus/blob/3fff8b56097778f1491edba968aee3cfbd5652ef/manifests/prometheusOperator-clusterRole.yaml#L13

**Comment by QuentinBisson** [comment]
Thank you @jkroepke

**Pull request merged**